libs/blocks/carousel/carousel.js

```diff
@@ -179,6 +180,47 @@ function setIndicatorMultiplyer(carouselElements, activeSlideIndicator, event) {
   }
 }
 
+function updateAriaLive(ariaLive, slide) {
+  let text = '';
+  slide.querySelectorAll(':scope > :not(.section-metadata').forEach((el, index) => {
+    text += `${index ? ' ' : ''}${el.textContent.trim()}`;
+  });
+  if (text) {
+    ariaLive.textContent = text;
+  } else {
+    const img = slide.querySelector('img[alt]');
+    const video = slide.querySelectorAll('video[title], iframe[title]');
+    ariaLive.textContent = img?.getAttribute('alt') || video?.getAttribute('title') || '';
+  }
+}
+
+function setAriaHiddenAndTabIndex(carouselElements, activeSlide) {
+  const { el: carouselBlock, slides } = carouselElements;
+  const active = activeSlide ?? carouselBlock.querySelector('.carousel-slide.active');
+  const activeSlideIndex = slides
+    .findIndex((el) => active === el);
+  const indexOfShowClass = [...carouselBlock.classList].findIndex((ele) => ele.includes('show-'));
+  let tempSlides = slides;
+  let noOfVisibleSlides = 1;
+  const mediaQueryMatches = window.matchMedia('(min-width: 900px)').matches;
+  if (indexOfShowClass >= 0 && mediaQueryMatches) {
+    noOfVisibleSlides = parseInt(carouselBlock.classList[indexOfShowClass].split('-')[1], 10);
+  }
+  if (activeSlideIndex > 0) {
+    tempSlides = [...slides.slice(activeSlideIndex), ...slides.slice(0, activeSlideIndex)];
+  }
+  tempSlides.forEach((slide, index) => {
+    let tabIndex = -1;
+    if (index < noOfVisibleSlides) {
+      tabIndex = 0;
+      slide.removeAttribute('aria-hidden');
+    } else {
+      slide.setAttribute('aria-hidden', true);
+    }
+    slide.querySelectorAll(FOCUSABLE_SELECTOR).forEach((focusableElement) => { focusableElement.setAttribute('tabindex', tabIndex); });
+  });
+}
```

Comment:
We can get more concise with the naming and structure ... An untested suggestion:
```suggestion
function setAriaHiddenAndTabIndex({ el: block, slides }, activeEl) {
    const active = activeEl ?? block.querySelector('.carousel-slide.active');
    const activeIdx = slides.findIndex(el => el === active);
    const isWide = window.matchMedia('(min-width: 900px)').matches;
    const showClass = [...block.classList].find(cls => cls.startsWith('show-'));
    const visible = isWide && showClass ? +showClass.split('-')[1] : 1;
    const ordered = activeIdx > 0 ? [...slides.slice(activeIdx), ...slides.slice(0, activeIdx)] : slides;
    ordered.forEach((slide, i) => {
      const isVisible = i < visible;
      slide.toggleAttribute('aria-hidden', !isVisible);
      slide.querySelectorAll(FOCUSABLE_SELECTOR).forEach(el =>
        el.setAttribute('tabindex', isVisible ? 0 : -1)
      );
    });
  }
```